lib/src/main/java/com/dynatrace/metric/util/DynatraceMetadataEnricher.java

```diff
@@ -39,48 +36,34 @@
    *     with empty key or value are discarded.
    */
   static List<Dimension> getDynatraceMetadata() {
-    return parseDynatraceMetadata(
-        getMetadataFileContentWithRedirection(
-            INDIRECTION_FILE_NAME, ALTERNATIVE_METADATA_FILENAME));
+    return createDimensionList(
+        getPropertiesWithIndirection(INDIRECTION_FILE_NAME, ALTERNATIVE_METADATA_FILENAME));
   }
 
   /**
-   * This function takes a list of strings from the Dynatrace metadata file and transforms it into a
-   * list of {@link Dimension} objects. Parsing failures will not be added to the output list.
-   * Therefore, it is possible that the output list is shorter than the input, or even empty.
+   * This function takes {@link Properties} object and transforms it into a {@link List<Dimension>}.
    *
-   * @param lines a {@link List<String>} containing key-value pairs (as one string) separated by an
-   *     equal sign.
+   * @param properties {@link Properties} to transform
    * @return A {@link List} of {@link Dimension dimensions} mapping {@link String} to {@link
-   *     String}. These represent the the lines passed in separated by the first occurring equal
-   *     sign on each line, respectively. If no line is parsable, returns an empty list. Dimensions
-   *     are not normalized.
+   *     String}. These represent the property entries, where empty keys or values were omitted.
+   *     Dimensions are not normalized.
    */
-  static List<Dimension> parseDynatraceMetadata(Collection<String> lines) {
+  static List<Dimension> createDimensionList(Properties properties) {
     ArrayList<Dimension> entries = new ArrayList<>();
 
-    // iterate all lines from metadata file.
-    for (String line : lines) {
-      logger.fine(String.format("parsing Dynatrace metadata: %s", line));
-      // if there are more than one '=' in the line, split only at the first one.
-      String[] split = line.split("=", 2);
+    for (Map.Entry<Object, Object> entry : properties.entrySet()) {
+      String key = entry.getKey().toString();
+      String value = entry.getValue().toString();
 
-      // skip if there is no '=' in the line
-      if (split.length != 2) {
-        logger.warning(String.format("could not parse metadata line ('%s')", line));
+      // skip if either key or value are empty
+      if (key.isEmpty() || value.isEmpty()) {
+        logger.log(Level.WARNING, () -> String.format("dropped settings %s=%s", key, value));
```

Comment:
Maybe say they were dropped because they are empty? And if they are empty does it make sense to log them?

Reply:
I added a reason to it as to why they were dropped.
I think it does make sense to log it. This way it is immediately obvious that something was dropped, and might even provide a hint as to which key is missing a value.